docs(react-scheduler): add several notes to docs (#2837)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/editing.md

## Basic Setup

To enable editing, follow the steps below:

1. Add the `EditingState` plugin and handle its `onCommitChanges` event to commit edits to the data storage.

> 2. Add either `IntegratedEditing` or `EditRecurrenceMenu` plugin.

Reviewer: How do I choose which plugin should I add?

Author: The `EditRecurrenceMenu` provides the capability to edit recurrent appointments, the `IntegratedEditing` does not. Should I clarify this? For example, "Add the `EditRecurrenceMenu` plugin if your data set may contain recurrent appoinments. Otherwise, add еру `IntegratedEditing` plugin"

Reviewer: ```suggestion
2. Add the `EditRecurrenceMenu` plugin if recurring appointments are allowed. Otherwise, add the `IntegratedEditing` plugin.
```

> 3. Add the `AppointmentForm` plugin to visualize appointment's data and to allow users to edit it.

Reviewer: ```suggestion
3. Add the `AppointmentForm` plugin so that users can edit appointment data.
```
